bokeh/io.py

```diff
@@ -424,6 +430,38 @@ def push(session_id=None, url=None, app_path=None, document=None, state=None, io
     _push_to_server(session_id=session_id, url=url, app_path=app_path,
                     document=document, io_loop=io_loop)
 
+def push_notebook(document=None, state=None):
+    ''' Update the notebook with the data for the current document.
+
+    Args:
+
+        document (Document, optional) : A :class:`bokeh.document.Document` to use
+
+        state (State, optional) : A state to use for any output_server() configuration of session or url
+
+    Returns:
+        None
+
+    '''
+    if state is None:
+        state = _state
+
+    if not document:
+        document = state.document
+
+    if not document:
+        warnings.warn("No document to push")
+        return
+
+    if not state.last_comms:
+        warnings.warn("Cannot find a last shown plot to update")
+        return
```

Comment:
this warning should probably say "call output_notebook() and show() before push_notebook()" or otherwise explain how to fix it